app/models/search.rb

```diff
@@ -23,7 +26,12 @@ def initialize
   end
 
   def max_matches
-    100
+    if SEARCH_MAX_PAGES.nil?
+      # A conservative maximum.
+      2 ** 31
+    else
+      SEARCH_MAX_PAGES * self.per_page
```

Comment:
I like how clear this multiplication is. Let's remove the `if` so it's clearer. Just stick it in your comment, `# Set this to 2 ** 31 for more than enough pages`. If someone's editing this, they know at least a little code, so we don't have to to offer them `nil` instead of directly using this constant.